src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/ActionBar.xaml.cs

```diff
@@ -48,6 +91,16 @@ private void ActionListViewItem_Tapped(object sender, TappedRoutedEventArgs e)
         if (listItem.DataContext is CommandContextItemViewModel item)
         {
             ViewModel?.InvokeItemCommand.Execute(item);
+            MoreCommandsButton.Flyout.Hide();
+            e.Handled = true;
         }
     }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("CodeQuality", "IDE0051:Remove unused private members", Justification = "VS has a tendency to delete XAML bound methods over-agressively")]
+    private void PrimaryButton_Tapped(object sender, TappedRoutedEventArgs e) =>
+        WeakReferenceMessenger.Default.Send<ActivateSelectedListItemMessage>();
```

Comment:
It was suggested that the WCT could have Behavior helpers for this for simple messages like this. Though not sure if records can be constructed in XAML either (would be nice). Ties into improvements Behaviors need anyway for directly supporting specific events like `Tapped` for AOT support anyway, but anyway, just wanted to leave this here for posterity of we could clean these up in the future to make it easier.